app/controllers/admin/merchants_controller.rb

```diff
@@ -6,10 +6,18 @@ def show
   end
 
   def update
-    user_to_be_updated = User.find(params[:id])
-    user_to_be_updated.disable_merchant_items
-    user_to_be_updated.update(role: 0)
-    flash[:notice] = "Merchant #{user_to_be_updated.name} downgraded"
-    redirect_to admin_user_path(user_to_be_updated)
+    if params[:disable]
+      merchant = User.find(params[:id])
```

Comment:
To dry this up, can we move line 10 above the if statement.  We could then delete line 16